File: Programmers/Lv.2/N개의 최소공배수/12953.java
class Solution {
    public int solution(int[] arr) {
        int answer = 1;
        
        for (int num : arr) {
            answer = lcm(answer, num);
        }
        
        return answer;
    }
    
    private int gcd(int a, int b) {
        return (b == 0) ? a : gcd(b, a % b);
    }
    
    private int lcm(int a, int b) {
        return (a * b) / gcd(a, b);
    }
}

File: Programmers/Lv.2/N개의 최소공배수/12953.js
function solution(arr) {
    let answer = 1;
    
    arr.forEach(v => {
        answer = lcm(answer, v);
    });
    
    return answer;
}

function gcd(a, b) {
    return (b === 0) ? a : gcd(b, a % b);
}

function lcm(a, b) {
    return (a * b) / gcd(a, b);
}
